### PR TITLE
SPY-1582 | Create declarative jenkins pipeline for spark-2.2

### PR DESCRIPTION
**Pull request by jayceeb**
Let's try using build/mvn

### CODE REVIEW

**Review by ianlcsd — APPROVED** [review]
new stuff that I don't know of.
since the change is only needed by csd, making the name csd specific?

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  stages{
+    stage('Checkout Source'){
+      steps{
+        checkout scm
+      }
+    }
+      stage('PR Builder'){
+      steps{
+        sh "build/mvn install"
```
Comment (jayceeb) [inline Jenkinsfile]: @markhamstra @ianlcsd would we need anything other than just a mvn install? or should i be including other flags

**Comment by markhamstra** [comment]
How are the maven command line options getting set (or maybe they are not)? Doing a single-pass build-and-test with `build/mvn -Pdeb -U -Phadoop-2.7 -Dhadoop.version=2.8.2 -Pkinesis-asl -Pyarn -Phive -Phive-thriftserver -Dpyspark -Dsparkr clean install` is supposed to work, but changes that break the single-pass logic have a tendency to creep in since the Apache PR-builder doesn't use maven. That means that a two-pass build, first with the addition of -DskipTests and then without skipTests and without clean, is more reliable.

File: Jenkinsfile (hunk 1 above)
Comment (markhamstra) [reply Jenkinsfile]: see other comment

**Comment by jayceeb** [comment]
Thanks @markhamstra I will change that

**Comment by ianlcsd** [comment]
so the following?

```
## build with -DskipTests
build/mvn -Pdeb -U -Phadoop-2.7 -Dhadoop.version=2.8.2 -Pkinesis-asl -Pyarn -Phive -Phive-thriftserver -Dpyspark -Dsparkr -DskipTests clean install

## then kick off test
build/mvn -Pdeb -U -Phadoop-2.7 -Dhadoop.version=2.8.2 -Pkinesis-asl -Pyarn -Phive -Phive-thriftserver -Dpyspark -Dsparkr test

```

**Comment by markhamstra** [comment]
@ian either `test` or `install` in the second pass will probably do the same thing, but `install` will also run any test suites that are under the `verify` goal -- not sure if Spark actually has any of those right now.

File: Jenkinsfile [hunk 2]
```diff
@@ -12,9 +12,14 @@ pipeline{
         checkout scm
       }
     }
-      stage('PR Builder'){
+    stage('Clean Install'){
       steps{
-        sh "build/mvn install"
+        sh "build/mvn -Pdeb -U -Phadoop-2.7 -Dhadoop.version=2.8.2 -Pkinesis-asl -Pyarn -Phive -Phive-thriftserver -Dpyspark -Dsparkr -DskipTests clean install"
```
Comment (jayceeb) [inline Jenkinsfile]: fyi @markhamstra @ianlcsd How i am injecting mvn flags

**Comment by ianlcsd** [comment]
ok, how about MAVEN_OPTS for MaxPermSize and Xmx setup? 
is it already part of build/mvn already?

**Comment by markhamstra** [comment]
@ianlcsd Java 8 doesn't use MaxPermSize. Also: https://github.com/clearstorydata/spark/blob/csd-2.2/build/mvn#L25 and https://github.com/clearstorydata/spark/blob/csd-2.2/build/mvn#L152

**Comment by ianlcsd** [comment]
great

**Comment by ianlcsd** [comment]
where/how is the target branch specified?

**Comment by jayceeb** [comment]
@ianlcsd The way I have it configured on Jenkins is that it will search for a  "Jenkinsfile" on any branch that I specify. If it finds one, it will create a build for it. 

http://jenkins.clearstorydata.com/view/Main/job/clearstorydata-github/configure

screen shot 2017-12-26 at 1 57 33 pm

**Pull request merged**